crypto/threads_pthread.c

```diff
@@ -57,10 +57,14 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
      * We don't use recursive mutexes, but try to catch errors if we do.
      */
     pthread_mutexattr_init(&attr);
-#  if !defined(NDEBUG) && !defined(OPENSSL_NO_MUTEX_ERRORCHECK)
+#  if defined (__TANDEM) && defined (_SPT_MODEL_)
+    /* SPT does not define MUTEX attributes. */
+#  else
```

Comment:
Why not `if !defined (__TANDEM) || !defined (_SPT_MODEL_)` and drop the else?

Reply:
Why not make it more general and use only `ifdef PTHREAD_MUTEX_xxx` as appropriate?